client/state/billing-data/util.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * External dependencies
+ */
+import { assign } from 'lodash';
+import { moment } from 'i18n-calypso';
+
+/**
+ * Parses the date within a transaction.
+ * Returns a copy of the updated transaction.
+ *
+ * @param  {Object}  transaction  Transaction object
+ * @return {Object}               Updated transaction with parsed date
+ */
+export const parseDate = ( transaction ) => {
+	return assign( {}, transaction, {
+		date: moment( transaction.date ).toDate()
```

Comment:
IndexedDB does support serializing date objects, but localStorage (a localForage fallback driver) will store this as a string. We can either have a custom serialize/deserialize method, or we can move this to the selector call.